src/plugins/dashboard/public/dashboard_app/no_data/dashboard_app_no_data.tsx

```diff
@@ -71,8 +74,28 @@ export const DashboardAppNoDataPage = ({
 export const isDashboardAppInNoDataState = async () => {
   const {
     data: { dataViews },
+    embeddable,
+    dashboardContentManagement,
+    dashboardBackup,
   } = pluginServices.getServices();
 
   const hasUserDataView = await dataViews.hasData.hasUserDataView().catch(() => false);
-  return !hasUserDataView;
+  if (hasUserDataView) return false;
+
+  // consider has data if there is an incoming embeddable
+  const hasIncomingEmbeddable = embeddable
```

Comment:
Changing the logic when to block the dashboard with no data views prompt. Needed to address this: 
https://github.com/elastic/kibana/pull/174316#issuecomment-1914722657

We need a dashboard to work in case a user without data views creates ES|QL chart in discover